xarray/core/variable.py

```diff
@@ -492,6 +495,46 @@ def _broadcast_indexes_advanced(self, key):
         key = VectorizedIndexer(variable.data for variable in variables)
         return dims, key
 
+    def _broadcast_indexes_dask_pointwise(self, key):
+        if any(not isinstance(k, (Variable, slice)) for k in key):
```

Comment:
@shoyer 
> Actually, this gives us a path to supporting vectorized indexing via the same code path for dask and numpy. 

That solution sounds much cleaner.
I am not familiar with under the hood of numpy, but the lookup cost does not change with `.vindex`?

> I don't have time to do that now but probably later this week.
Thanks.
I will take a deeper look at the other part, such as `.sel` and `.sel_points`.